test/test-matrix-api-r0.3.0.el

```diff
@@ -0,0 +1,106 @@
+(require 'cl-lib)
```

Comment:
I'm really happy that you created tests, but I'm a bit worried that running them would be a bit hard, with setting up credentials and querying to a live matrix server. 

Maybe we could mock some of the functions to provide more of a 'unit test', but that would be less complete. I don't think it's worth persuing right now, but I might look into that later on.

Reply:
Well, it currently uses a static account on the matrix.org server, and only requires that you have a file `password.el` in the repo directory with `(setq password "password")` (and I will give you the password to the test account over a private channel whenever you're ready).

It might make some things simpler to mock them, but I feel like that might also lean toward reimplementing a matrix server, haha.  On the other hand, if we record some valid responses to the tests, we could hard-code-mock those into the test, which might be a good idea.